### PR TITLE
Cancel concurrent CI runs for branches other than main

### LINKED ISSUE #298: Cancel concurrent CI runs
https://www.meziantou.net/how-to-cancel-github-workflows-when-pushing-new-commits-on-a-branch.htm

### PR DESCRIPTION
**Pull request by kscharan**
Closes #298

### CODE REVIEW

**Comment by gussmith23** [comment]
Closes #298

File: .github/workflows/run-checks-github-runners.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ env:
   # Can't use ${{ env.REGISTRY }} here, but would like to, so that we' don't
   # repeat ghcr.io.
   IMAGE_TAG: ghcr.io/${{ github.repository }}:sha-${{ github.sha }}
+  
+concurrency:
+  # Cancel in-progress runs for branches other than main
```
Comment (gussmith23) [inline .github/workflows/run-checks-github-runners.yml:23]: I don't think this comment is exactly correct. This doesn't necessarily _ignore_ the main branch. It just uses a different group ID if the branch is main (it uses run_id if the branch is main). This basically means that this workflow won't get canceled for pushes to main, which is fine for our purposes.

Can you add a reference to the source where you took the code from?
https://www.meziantou.net/how-to-cancel-github-workflows-when-pushing-new-commits-on-a-branch.htm

Thank you!

**Review by gussmith23 — CHANGES_REQUESTED** [review]
See comment! Thank you so much!

File: .github/workflows/run-checks-github-runners.yml (hunk 1 above)
Comment (gussmith23) [reply .github/workflows/run-checks-github-runners.yml:23]: To be clear, please:

- [ ] Update the comment to reflect what the concurrency group actually does
- [ ] Add a reference to the link

**Comment by kscharan** [comment]
Hello @gussmith23 , 
I have changed the approach. Can you please re-review it?

**Pull request merged**